overlord/state/task.go

```diff
@@ -159,3 +163,16 @@ func (t *Task) Get(key string, value interface{}) error {
 	t.state.ensureLocked()
 	return t.data.get(key, value)
 }
+
+// WaitFor registers another task as a requirement for t to make progress
+// and sets the status as WaitingStatus.
+func (t *Task) WaitFor(another *Task) {
+	t.state.ensureLocked()
+	t.status = WaitingStatus
+	t.waitTasks = append(t.waitTasks, another.ID())
+}
+
+// WaitTasks returns the list of tasks registered for t to wait for.
+func (t *Task) WaitTasks() []string {
```

Comment:
it means also it needs t.state.ensureLocked(), that's why btw I ended up writing tests about that being present for all my bits in the end